src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -247,6 +265,27 @@ public CodeWriter emit(CodeBlock codeBlock) throws IOException {
           break;
 
         default:
+          if (potentiallyStaticImported == null) {
+            emitAndIndent(part);
+            break;
+          }
+          ClassName pendingClassName = potentiallyStaticImported;
```

Comment:
I think you want a helper method here. That way you can avoid the duplication between 268 and 288 ?